### ISSUE TITLE
Generate current_courses.php on the fly

### ISSUE THREAD

**Issue by zjxiaohan** [Submitty/Submitty#3914]
**What problem are you trying to solve with Submitty**
`current_courses.php` is not dynamically generated.

**Describe the way you'd like to solve this problem**
Enable Submitty to generate a list of current courses in the fly so that `ErrorView.php` does not need to get data from `current_courses.php` in the future.

**Additional context**
Thank @sak6lab @MasterOdin @prestoncarman for extending the discussion PR #3902 .

**Comment by MasterOdin** [comment]
Closing this as it's not really the solution we want. I talked to @bmcutler today about this and have a half done issue on what we'd like to do this page that I'll finish tomorrow morning. This page itself exists purely for historical reasons that are no longer true, and so there's no point spending time trying to change parts of it, when it should just go away.

The gist of my wip issue is that this page should go away completely as it stands, and instead, display a page similar to the one you get when you attempt to go to a course you don't have access to, which shows an error, but leaves in place all the appropriate navigation elements, which is better UI.
